challenges/2021-06-08-simple-2d-iterator/solutions/java/codingis4noobs2/Codingis4noobs2.java

```diff
@@ -0,0 +1,47 @@
+// 2D Array Itreator
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+class NestedIterator<T> implements Iterator<T> {
+    private T[][] array;
+    private int row;
+    private int col;
+
+    public NestedIterator(T[][] array) {
+        this.array = array;
+        row = 0;
+        col = 0;
+    }
+
+    public boolean hasNext() {
+        return row < array.length && col < array[row].length;
+    }
+
+    public T next() {
+        if (!hasNext()) {
+            throw new NoSuchElementException();
+        }
+        T result = array[row][col];
+        col++;
+        if (col == array[row].length) {
+            row++;
+            col = 0;
+        }
+        return result;
+    }
+}
+
+public class Codingis4noobs2 {
+    public static void main(String[] args) {
```

Comment:
Nothing "wrong" with demonstrating your solution this way, but I'm going to add a comment to the PR demonstrating how to incorporate and test your code with JUnit. It'll probably look a bit wonky since it'll be using JUnit directly (instead of through an IDE or using a package manager), but it's worth taking a look at. I'm not going to ask that you change your PR to use JUnit, just take a look.